animations/test/misc/carousel_test.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/misc/carousel.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+Widget createCarouselDemoScreen() => MaterialApp(
+      home: CarouselDemo(),
+    );
+
+void main() {
+  testWidgets('CarouselDemo swipe test', (tester) async {
+    await tester.pumpWidget(createCarouselDemoScreen());
+
+    // Get the images available on the screen during initial state.
+    var imageList = tester.widgetList(find.byType(Image)).toList();
+    expect(imageList.length, 2);
+
+    // Swipe the Carousel.
+    await tester.fling(find.byType(CarouselDemo), Offset(-400, 0), 800);
```

Comment:
Nice that you're targeting the swipe on the main widget, rather than digging into it. Makes this test slightly less coupled to the internals.